xds/src/main/java/io/grpc/xds/sds/TlsCertificateStore.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds;
+
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+import io.envoyproxy.envoy.api.v2.auth.TlsCertificate;
+import io.envoyproxy.envoy.api.v2.core.DataSource;
+import io.grpc.Internal;
+import java.io.InputStream;
+
+/**
+ * TlsCertificate's PrivateKey and Certificate are extracted into InputStream's.
+ * This is used by the gRPC SSLContext/Protocol Negotiator and is internal.
+ * See {@link SecretManager} for a note on lifecycle management.
+ */
+@Internal
+public final class TlsCertificateStore {
+
+  private final InputStream privateKeyStream;
+  private final InputStream certChainStream;
+
+  private static InputStream getInputStreamFromDataSource(DataSource dataSource) {
+    Preconditions.checkNotNull(dataSource);
+    ByteString dataSourceByteString = null;
+    if (dataSource.getSpecifierCase() == DataSource.SpecifierCase.INLINE_BYTES) {
+      dataSourceByteString = dataSource.getInlineBytes();
+    } else if (dataSource.getSpecifierCase() == DataSource.SpecifierCase.INLINE_STRING) {
+      dataSourceByteString = dataSource.getInlineStringBytes();
+    } else {
+      throw new IllegalArgumentException(
```

Comment:
nit: UnsupportedOperationException?

Reply:
hmmm... debatable. Illegal argument also makes sense since `dataSource.getSpecifierCase()` is checked and based on that this is thrown. But I will change it (unless you are convinced by my statement!)

Reply:
i am fine either way. i just noticed that the message mentions unsupported.